api/internals/usecase/purchase_order_usecase.go

```diff
@@ -71,68 +69,94 @@ func (p *purchaseOrderUseCase) GetPurchaseOrderByID(c context.Context, id string
 	return purchaseOrder, nil
 }
 
-//PurcahseOrderの作成(create)
+// PurcahseOrderの作成(create)
 func (p *purchaseOrderUseCase) CreatePurchaseOrder(
-	c context.Context, 
-	DeadLine string,
+	c context.Context,
+	deadLine string,
 	userID string,
-	FinanceCheck string,
-) error {
-	err := p.rep.Create(c, DeadLine, userID, FinanceCheck)
-	return err
+	finansuCheck string,
+) (domain.PurchaseOrder, error) {
+	latastPurchaseOrder := domain.PurchaseOrder{}
+	p.rep.Create(c, deadLine, userID, finansuCheck)
+	row, err := p.rep.FindNewRecord(c)
+	err = row.Scan(
+		&latastPurchaseOrder.ID,
+		&latastPurchaseOrder.DeadLine,
+		&latastPurchaseOrder.UserID,
+		&latastPurchaseOrder.FinanceCheck,
+		&latastPurchaseOrder.CreatedAt,
+		&latastPurchaseOrder.UpdatedAt,
+	)
+	if err != nil {
+		return latastPurchaseOrder, err
+	}
+	return latastPurchaseOrder, nil
 }
 
-//PurchaseOrderの修正(Update)
+// PurchaseOrderの修正(Update)
 func (p *purchaseOrderUseCase) UpdatePurchaseOrder(
 	c context.Context,
 	id string,
-	DeadLine string,
+	deadLine string,
 	userID string,
-	FinanceCheck string,
-) error {
-	err := p.rep.Update(c, id, DeadLine, userID, FinanceCheck)
-	return err
+	finansuCheck string,
+) (domain.PurchaseOrder, error) {
+	updatedPurchaseOrder := domain.PurchaseOrder{}
+	p.rep.Update(c, id, deadLine, userID, finansuCheck)
+	row, err := p.rep.Find(c, id)
+	err = row.Scan(
+		&updatedPurchaseOrder.ID,
+		&updatedPurchaseOrder.DeadLine,
+		&updatedPurchaseOrder.UserID,
+		&updatedPurchaseOrder.FinanceCheck,
+		&updatedPurchaseOrder.CreatedAt,
+		&updatedPurchaseOrder.UpdatedAt,
+	)
+	if err != nil {
+		return updatedPurchaseOrder, err
+	}
+	return updatedPurchaseOrder, nil
 }
 
-//PurchaseOrderの削除(delete)
+// PurchaseOrderの削除(delete)
 func (p *purchaseOrderUseCase) DestroyPurchaseOrder(
 	c context.Context,
 	id string,
-)error {
+) error {
 	err := p.rep.Delete(c, id)
 	return err
 }
 
-//Purchase_orderに紐づくUserとItemの取得(All)
-func (p *purchaseOrderUseCase) GetOrderWithUserItem(c context.Context) ([]domain.OrderWithItemAndUser,error){
-	orderWithUserAndItem := domain.OrderWithItemAndUser{}
-	var orderWithUserAndItems []domain.OrderWithItemAndUser
+// Purchase_orderに紐づくUserとItemの取得(All)
+func (p *purchaseOrderUseCase) GetPurchaseOrderDetails(c context.Context) ([]domain.OrderDetail, error) {
+	orderDetail := domain.OrderDetail{}
+	var orderDetails []domain.OrderDetail
 	purchaseItem := domain.PurchaseItem{}
 	var purchaseItems []domain.PurchaseItem
 	rows, err := p.rep.AllOrderWithUser(c)
```

Comment:
```suggestion
	rows, err := p.rep. AllUserInfo(c)
```
repositoryの修正に倣うとこんな感じになる